<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>704</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>704</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>RE Proxy - multi protocol reversing proxy</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLineEdit" name="eLPort">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>local port number to open for receiving connections from the client.</string>
    </property>
    <property name="text">
     <string>12345</string>
    </property>
    <property name="cursorPosition">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="eRHost">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>30</y>
      <width>251</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>127.0.0.1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bConnect">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>20</y>
      <width>171</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>local port:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>10</y>
      <width>91</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>remote host:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>10</y>
      <width>81</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>remote port:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eRPort">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>30</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>8080</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>0</y>
      <width>71</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Protocol:</string>
    </property>
    <widget class="QRadioButton" name="rTCP">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>51</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>TCP</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="rUDP">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>UDP</string>
     </property>
    </widget>
   </widget>
   <widget class="QCheckBox" name="chkAutoSend">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Auto send </string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="bSend">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>60</y>
      <width>491</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>&gt;&gt;&gt; Send &gt;&gt;&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>510</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>IN:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eIn">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>510</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>510</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>OUT:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eOut">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>510</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>510</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Size:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eSize">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>510</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>510</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Mutation:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eMutation">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>510</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>510</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>%</string>
    </property>
   </widget>
   <widget class="QLabel" name="lStatus">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>540</y>
      <width>621</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Dissconnected</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>510</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>ID:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="eId">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>510</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QTableWidget" name="tHex">
    <property name="geometry">
     <rect>
      <x>5</x>
      <y>101</y>
      <width>691</width>
      <height>401</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>1</width>
      <height>0</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Unifont</family>
      <pointsize>11</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>704</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Hex"/>
    <addaction name="actionLoad_Bin"/>
    <addaction name="actionSave_Hex"/>
    <addaction name="actionSave_Bin"/>
    <addaction name="actionSave_all"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Search</string>
    </property>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="actionSearch_Text"/>
    <addaction name="actionSearch_hexa"/>
    <addaction name="actionReplace_text"/>
    <addaction name="actionReplace_hexa"/>
   </widget>
   <widget class="QMenu" name="menuScripts">
    <property name="title">
     <string>Scripts</string>
    </property>
    <addaction name="actionLoad_script"/>
   </widget>
   <widget class="QMenu" name="menuTools_2">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionRadare"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionAbout_2"/>
    <addaction name="actionScripting"/>
   </widget>
   <widget class="QMenu" name="menuCrypto">
    <property name="title">
     <string>Crypto</string>
    </property>
    <addaction name="actionchacha20"/>
    <addaction name="actionAES"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuScripts"/>
   <addaction name="menuTools_2"/>
   <addaction name="menuCrypto"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionLoad_Hex">
   <property name="text">
    <string>Load Hex</string>
   </property>
  </action>
  <action name="actionLoad_Bin">
   <property name="text">
    <string>Load  Bin</string>
   </property>
  </action>
  <action name="actionSave_Hex">
   <property name="text">
    <string>Save Hex</string>
   </property>
  </action>
  <action name="actionSave_Bin">
   <property name="text">
    <string>Save Bin</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionManual">
   <property name="text">
    <string>Manual</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionAbout_2">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSave_all">
   <property name="text">
    <string>Save all</string>
   </property>
  </action>
  <action name="actionRadare">
   <property name="text">
    <string>Radare</string>
   </property>
  </action>
  <action name="actionLoad_script">
   <property name="text">
    <string>Load script</string>
   </property>
  </action>
  <action name="actionScripting">
   <property name="text">
    <string>Scripting</string>
   </property>
  </action>
  <action name="actionSearch_Text">
   <property name="text">
    <string>Search Text</string>
   </property>
  </action>
  <action name="actionSearch_hexa">
   <property name="text">
    <string>Search hexa</string>
   </property>
  </action>
  <action name="actionReplace_text">
   <property name="text">
    <string>Replace text</string>
   </property>
  </action>
  <action name="actionReplace_hexa">
   <property name="text">
    <string>Replace hexa</string>
   </property>
  </action>
  <action name="actionchacha20">
   <property name="text">
    <string>chacha20</string>
   </property>
  </action>
  <action name="actionAES">
   <property name="text">
    <string>AES</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>eLPort</tabstop>
  <tabstop>eRHost</tabstop>
  <tabstop>eRPort</tabstop>
  <tabstop>bConnect</tabstop>
  <tabstop>rTCP</tabstop>
  <tabstop>rUDP</tabstop>
  <tabstop>chkAutoSend</tabstop>
  <tabstop>bSend</tabstop>
  <tabstop>eIn</tabstop>
  <tabstop>eOut</tabstop>
  <tabstop>eSize</tabstop>
  <tabstop>eMutation</tabstop>
  <tabstop>eId</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
